packages/booter-lb3app/fixtures/app-with-model.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const loopback = require('loopback');
+const boot = require('loopback-boot');
+
+const app = (module.exports = loopback());
+
+app.dataSource('memory', {connector: 'memory'});
+const Color = app.registry.createModel('Color', {name: String});
+app.model(Color, {dataSource: 'memory'});
```

Comment:
Can we also add a non-in-memory datasource test case?

Reply:
AFAIK, it won't add value to the test, it will rather add a dependency. We just want to check if a connector is added to the context at all.

Thoughts @strongloop/loopback-next?

Reply:
What if we add the test in `repository-test` folder? Will it be able to test if the model is migrated to mongoDB or MySQL? It's part of the criteria. 
I am ok to skip the test if @strongloop/loopback-maintainers think it's fine.

Reply:
Is this PR aimed to fix all issues in #2825? IIUC there will be another PR to make sure the `npm run migrate` migrates LB3 models and that is where we can test different datasources.

I feel the test here is just to make sure lb3 datasources are loaded properly and bound to the lb4 app, so it doesn't matter which database it uses :)

Reply:
Here is my thinking:

- This pull request is the complete implementation of #2825
- We trust the LB3 runtime to correctly migrate LB3 models when `dataSource.automigrate` is called, this does not need to be tested in LB4 again.
- What we need to ensure and test: `npm run migrate` will call `automigrate`/`autoupdate` on datasources defined in the LB3 app.
- Yaapa run a manual test to verify that `npm run migrate` migrates LB3 models.

> I feel the test here is just to make sure lb3 datasources are loaded properly and bound to the lb4 app, so it doesn't matter which database it uses :)

+1

> What if we add the test in `repository-test` folder?

These tests are verifying `@loopback/repository` functionality. In this pull request, we want to test integration between LB3 and LB4, I feel that does not belong to `@loopback/repository-test`.

We can add a new package to the top-level `acceptance` directory, alongside `repository-mysql` and `repository-mongodb`. However, I feel the cost of running such acceptance test is not worth the benefits. It should be the responsibility of LB3 to test and verify that automigration works as expected.

Reply:
I see, thank you for the clarification!